test/extrema_test.dart

```diff
@@ -0,0 +1,112 @@
+      Logic(width: 8)..put(LogicValue.ofString('00001111')) // 15 in decimal
+    ];
+
+    // Create an instance of FindMin.
+    final findMin = Extrema(logics, max: false);
+    await findMin.build();
+
+    // Create an instance of FindMax.
+    final findMax = Extrema(logics);
+    await findMax.build();
+
+    // Verify the min value and index
+    expect(findMin.val.value.toInt(), equals(3));
+    expect(findMin.index.value.toInt(), equals(3));
+
+    // Verify the max value and index.
+    expect(findMax.val.value.toInt(), equals(109));
+    expect(findMax.index.value.toInt(), equals(0));
+  });
+
+  test('Extrema of a list of Logics, different widths.', () async {
+    // Create a list of Logic objects with different values.
+    final logics = [
+      Logic(width: 8)..put(LogicValue.ofString('01101101')), // 109 in decimal
+      Logic(width: 4)..put(LogicValue.ofString('0101')), // 5 in decimal
+      Logic(width: 8)..put(LogicValue.ofString('00010100')), // 20 in decimal
+      Logic(width: 2)..put(LogicValue.ofString('11')), // 3 in decimal
+      Logic(width: 8)..put(LogicValue.ofString('00001111')) // 15 in decimal
+    ];
+
+    // Create an instance of FindMin.
+    final findMin = Extrema(logics, max: false);
+    await findMin.build();
+
+    // Create an instance of FindMax.
+    final findMax = Extrema(logics);
+    await findMax.build();
+
+    // Verify the min value and index
+    expect(findMin.val.value.toInt(), equals(3));
+    expect(findMin.index.value.toInt(), equals(3));
+
+    // Verify the max value and index.
+    expect(findMax.val.value.toInt(), equals(109));
+    expect(findMax.index.value.toInt(), equals(0));
+  });
+
+  test('List containing one element.', () async {
+    // Create a list of Logic objects with different values
+    final logics = [
+      Logic(width: 4)..put(LogicValue.ofString('1100')), // 12 in decimal
+    ];
+
+    // Create an instance of FindMin.
+    final findMin = Extrema(logics, max: false);
+    await findMin.build();
+
+    // Create an instance of FindMax
+    final findMax = Extrema(logics);
+    await findMax.build();
+
+    // Verify the minimum value and index
+    expect(findMax.val.value.toInt(), equals(12));
+    expect(findMax.index.value.toInt(), equals(0));
+  });
+
+  test('List containing an empty element.', () async {
+    // Create a list of Logic objects with different values
+    final logics = [Logic(width: 4)]; // Empty element
+
+    // Create an instance of FindMax
+    final findMax = Extrema(logics);
+    await findMax.build();
+
+    // Verify the minimum value and index
+    expect(() => findMax.val.value.toInt(), throwsA(isA<RohdException>()));
+    expect(findMax.index.value.toInt(), equals(0));
+  });
+
+  test('List containing no elements.', () async {
+    // Create a list of Logic objects with
+    List<Logic> logics = [];
+
+    // Try to create an instance of Extrema
+    expect(() => Extrema(logics), throwsA(isA<ArgumentError>()));
```

Comment:
good test, but we should throw a more helpful error message like a `RohdHclException` explaining that you can't have an empty list